### ISSUE TITLE
access forbidden error

### ISSUE THREAD

**Issue by alairs** [h2o/h2o#419]
### My h2o.conf
hosts:
  "res.ltecn.com":
    listen:
      port: 80
    paths:
      "/":
        file.dir: /root/res/
access-log: /root/h2olog/access-log
error-log: /root/h2olog/error-log
#pid-file: /root/h2olog/pid-file
http2-reprioritize-blocking-assets: ON   # performance tuning option

the file dir

root@bwgvps:~/res# ls -lh
total 3.3M
-rw-r--r--  1 root root  18K Jul 22 04:26 FileZilla.xml
drwxr-xr-x  4 root root 4.0K Jul 22 09:29 Font-Awesome-3.2.1
drwxr-xr-x  2 root root 4.0K Jul 21 23:53 Heart_Sutra
drwxr-xr-x  2 root root 4.0K Jul 21 23:54 KoreanEnt
drwxr-xr-x 16 root root 4.0K Jul 21 23:54 LFS-BOOK-7.7-systemd
-rw-r--r--  1 root root  13K Jul 22 04:26 LTE_Bands_LTE_Frequency.htm
drwxr-xr-x  3 root root 4.0K Jul 21 23:54 alinstall
drwxr-xr-x  2 root root 4.0K Jul 21 23:56 apks
drwxr-xr-x  5 root root 4.0K Jul 21 23:52 bootstrap-3.3.5-dis

**Comment by kazuho** [comment]
Is `/root/res` directory (or `/root` directory) configured to be readable by users other than root?

As described in https://h2o.examp1e.net/configure/base_directives.html#user, when spawned with root privileges H2O automatically setuids to nobody (you would also find a warning in error log that it is doing so).  Such behavior is considered necessary for a daemon process to reduce security risks.

In other words, the directory to be served by H2O should be configured to be readable by `nobody` (or the user specified using the `user` directive), by using the `chmod` or any other command.

**Comment by alairs** [comment]
@kazuho  

	#chmod -R 755 /root/res/
	#chmod -R root:root /root/res
	#h2o -m daemon -c h2o.conf

still access forbidden

**Comment by kazuho** [comment]
Isn't `/root` set to 0700?  IIRC that is the default for most Unix-based operating systems.

**Comment by HLFH** [comment]
@kazuho Hi, a guy named @Angristan also had this bug when file.dir was pointing to a folder in /root/ with www-data user rights as www-data was specified in /etc/h2o.conf. When he moved to /var/www/, everything was ok. I'm not sure it's really a bug, and I'm **not sure that /root/ is a recommended place for the web server**.

**Comment by kazuho** [comment]
@HLFH Thank you for the info.  It might be the case that SELinux (or some other security module) is prohibiting access by h2o to `/root`.  If that is the case, you should be able to find an audit error emitted to syslog.

**Comment by Varbin** [comment]
FWIW: If the files itself has read permissions, but h2o shows 403, it might help to give all parent folders the "x" permission (means search permission on folders). This  allows h2o to check if the file requested is present (if h2o isn't allowed to check it shows 403).

**Comment by rajhlinux** [comment]
When accessing the website I get "access forbidden" 

Using internal or external IP address.
Using the port 443.

When I use any port above 1024, I don't get the "access forbidden"

How can I use port 443 without getting the "access forbidden" when visiting the website hosted by H2O?

Thanks.
